File: benchmarks/struct_records/struct_records.py
class Person:
    __slots__ = ("name", "age")

    def __init__(self, name, age):
        self.name = name
        self.age = age


def make_person(i):
    return Person("user" + str(i), 20 + i % 50)


def birthday(p):
    return Person("user" + str(p.age), p.age + 1)


def score(p):
    return len(p.name) + p.age


def main():
    total = 0
    for i in range(500000):
        p = make_person(i)
        q = birthday(p)
        total += score(q)
    assert total == 25750000, "struct_records checksum"
    print("assert passed, struct_records is correct")


main()

File: benchmarks/struct_records/struct_records.go
package main

import "fmt"

type Person struct {
	Name string
	Age  int
}

func makePerson(i int) Person {
	return Person{Name: fmt.Sprintf("user%d", i), Age: 20 + i%50}
}

func birthday(p Person) Person {
	return Person{Name: p.Name, Age: p.Age + 1}
}

func score(p Person) int {
	return len(p.Name) + p.Age
}

func main() {
	total := 0
	for i := range 500000 {
		p := makePerson(i)
		q := birthday(p)
		total += score(q)
	}
	if total != 27638890 {
		panic("struct_records checksum")
	}
	fmt.Println("assert passed, struct_records is correct")
}
